### DISCUSSION TITLE
Possible to become root within blink?

### DISCUSSION THREAD

**Discussion by de-wim** [jart/blink#141] [Q&A]
I'm trying to run a Linux chroot on Mac.

I've grabbed a stage3 Gentoo tarball built against musl, modified `/etc/shadow`, and used `blink -C gentoo-musl/ gentoo-musl/bin/bash`to get a shell in the chroot. This seems to work, however, I'm unable to obtain root permissions by using `su -`.\

Did I do something wrong with my shadow file, or is this not possible in blink?

**Comment by tkchia** [comment]
Hello @wvdschel,

> This seems to work, however, I'm unable to obtain root permissions by using `su -`.

I guess, for this to work, one will need to modify Blink so that it can be installed, or at least run, as a setuid program &mdash; without creating too much of a security loophole.

On a non-emulated Linux system, `su` is normally installed with a setuid bit in its permissions:
```
$ ls -l /bin/su
-rwsr-xr-x 1 root root 55672 Feb 21  2022 /bin/su
```
The setuid bit, plus the fact that it is owned by `root`, are what give it the power to switch to another user id.

Thank you!
